app/src/test/java/com/example/marvelapp/data/mapper/CharacterMapperTest.kt

```diff
@@ -0,0 +1,211 @@
+package com.example.marvelapp.data.mapper
+
+import com.example.marvelapp.data.service.response.CharacterResponse
+import com.example.marvelapp.data.service.response.ComicsResponse
+import com.example.marvelapp.data.service.response.EventsResponse
+import com.example.marvelapp.data.service.response.SeriesResponse
+import com.example.marvelapp.data.service.response.StoriesResponse
+import com.example.marvelapp.data.service.response.ThumbnailResponse
+import com.example.marvelapp.data.service.response.UrlResponse
+import com.example.marvelapp.domain.entity.CharacterDetailEntity
+import com.example.marvelapp.domain.entity.CharacterEntity
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+
+class CharacterMapperTest {
+    private val mockCharacterResponse = mock(CharacterResponse::class.java)
+    private val mockThumbnailResponse = mock(ThumbnailResponse::class.java)
+    private val mockComicsResponse = mock(ComicsResponse::class.java)
+    private val mockUrlResponse = mock(UrlResponse::class.java)
+    private val mockSeriesResponse = mock(SeriesResponse::class.java)
+    private val mockStoriesResponse = mock(StoriesResponse::class.java)
+    private val mockEventsResponse = mock(EventsResponse::class.java)
+    private lateinit var mapper: CharacterMapper
+
+    @Before
+    fun setUp() {
+        mapper = CharacterMapper
+    }
+
+    @Test
+    fun validValuestest() {
+        whenever(mockCharacterResponse.id).thenReturn(ID)
```

Comment:
dont create a mock of characterResponse and mock every propertie.

just us a real characterResponse.

Reply:
yes, it is not a good idea mock data classes

Reply:
please check the in the rest of test. you should not mock a data class

Reply:
Done thx guys